src-jdk8/orchard/java/legacy_parser.clj

```diff
@@ -273,6 +273,10 @@
         (assoc (->> (map parse-info (.classes root))
                     (filter #(= klass (:class %)))
                     (first))
+               ;; relative path on the classpath
                :file path
-               :path (. (io/resource path) getPath))))
+               ;; Legacy key. Please do not remove - we don't do breaking changes!
+               :path (-> path io/resource .getPath)
```

Comment:
mmm now that I added :path again in the second commit, there are two io/resource calls, which go against this PR's intent.

Can extract to a common `let` later today (if you don't beat me to it)

Reply:
I've added a longer comment, but I really think this should simply go away or become an alias for `:resource-url`. Again I'll say that we shouldn't go overboard for backwards compatibility concerns that likely don't have any real-world implications. There are like 5 projects using `cider-nrepl` and all of them model their API usage on CIDER, as I never documented the API particularly well. :-)

Reply:
What's `overboard` here? The code is pretty simple, I'm not contorting it to keep compat (see `Don't compute io/resource twice for :path` commit).

In terms of computations, no extra cost is incurred. It's Just Data 🙂

Honestly I don't understand why the default should be dropping instead of keeping. In this case, dropping is being actually costlier as several people have to be queried/coordinated to make sure it's safe.

And we can't never be really sure. Clojure/Emacs are hackers' tools they can be consumed in unsuspected ways.

Reply:
For me this PR had two goals:

- eliminate a performance problem
- make the API simpler

If we still compute a redundant key, that's one redundant computation, even after the optimization you did. Not a big deal perhaps. More importantly, however, it's one key with unclear purpose in the result that was added somewhat accidentally. I mean - even I couldn't tell you what this key was about, before looking into the code. Now we're back to square one - the key is here, and people will wonder what exactly is the difference between all the "path" keys. Anyways, I don't want to argue too much on this topic and I'll merge the PR in its current form for now. I might end up simplifying this API myself down the road.